src/python/pants/backend/jvm/targets/scala_library.py

```diff
@@ -82,3 +89,37 @@ def java_sources(self):
         raise TargetDefinitionException(self, 'No such java target: {}'.format(spec))
       targets.append(target)
     return targets
+
+  @property
+  def scalac_plugins(self):
+    """The names of compiler plugins to use when compiling this target with scalac.
+    :return: See constructor.
+    :rtype: list of strings.
+    """
+    if self._scoverage:
+      return ScoveragePlatform.global_instance().get_scalac_plugins(self)
+
+    return self.payload.scalac_plugins
+
+  @property
+  def scalac_plugin_args(self):
```

Comment:
There could be better decoupling here rather than checking `if self._scoverage`,
possible suggestion could be something similar to: (plugins, may not not need to be passed in but could be a class variable e.g.)
```
def scalac_plugin_args(self, plugins):
for plugin in plugins:
  # supplement args
```